### DISCUSSION TITLE
Using ruleset templates with new mod overlay throws null reference exception

### DISCUSSION THREAD

**Discussion by Joppe27** [ppy/osu#18710] [Q&A] [answered]
What I did:
1. I built the ruleset-scrolling-example template
2. I put the resulting .dll file in the rulesets folder as I do normally
3. I ran the osu! Desktop project in debug

When I switch to the osu!coins ruleset in song select, the following nullref gets thrown:
```
System.NullReferenceException
  HResult=0x80004003
  Message=Object reference not set to an instance of an object.
  Source=osu.Game
  StackTrace:
   at osu.Game.Overlays.Mods.ModSelectOverlay.<>c.<createLocalMods>b__36_0(Mod mod) in C:\Users\joppe\osu\osu.Game\Overlays\Mods\ModSelectOverlay.cs:line 292
```
Or in video form: https://joppe27.s-ul.eu/YwriOXfh

To try and make sure I wasn't messing something up, I did a checkout on 885a285 (release 2022.205.0) which is before the new mod screen was implemented (since the stack trace leads to the mod overlay). Following the same steps the template does work as expected.

Not experienced so sadly can't say what the underlying issue might be, but thought it was worth reporting.

**Comment by bdach** [comment]
https://github.com/ppy/osu/issues/18298

the ruleset template [does the same wrong thing](https://github.com/ppy/osu/blob/e880fcf3cb57f8be0f7a40754bd94da6cd5a83c0/Templates/Rulesets/ruleset-scrolling-example/osu.Game.Rulesets.Pippidon/PippidonRuleset.cs#L37)

**Comment by bdach** [comment, marked as answer]
will be rectified by https://github.com/ppy/osu/pull/18715 if merged
